### PR TITLE
python37: Remove references to openssl.dev

### LINKED ISSUE #51465: python37 has a runtime dependency on openssl-dev
## Issue description
```
$ nix-store -qR $(nix-build -A python3) | grep dev
/nix/store/45ji26zar3sf62b0ahvly0y8a1szad8r-openssl-1.0.2p-dev
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

### PR DESCRIPTION
**Pull request by andrew-d**
###### Motivation for this change
Updates the `postInstall` to also strip the configure flag that contains `--with-openssl=/nix/store/...`, which results in a reference to `openssl.dev`
Fixes #51465

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @FRidh

### CODE REVIEW

File: pkgs/development/interpreters/python/cpython/3.7/default.nix [hunk 1]
```diff
@@ -145,7 +145,11 @@ in stdenv.mkDerivation {
     # some $TMPDIR references to improve binary reproducibility.
     # Note that the .pyc file of _sysconfigdata.py should be regenerated!
     for i in $out/lib/python${majorVersion}/_sysconfigdata*.py $out/lib/python${majorVersion}/config-${majorVersion}m*/Makefile; do
-      sed -i $i -e "s|-I/nix/store/[^ ']*||g" -e "s|-L/nix/store/[^ ']*||g" -e "s|$TMPDIR|/no-such-path|g"
+      sed -i $i \
+        -e "s|-I/nix/store/[^ ']*||g" \
```
Comment (dezgeg) [inline pkgs/development/interpreters/python/cpython/3.7/default.nix]: I know this is an existing problem, but we shouldn't assume that the Nix store is at `/nix/store`.

Comment (Ericson2314) [reply pkgs/development/interpreters/python/cpython/3.7/default.nix]: `NIX_BUILD_TOP` is the best thing to use instead?

Comment (andrew-d) [reply pkgs/development/interpreters/python/cpython/3.7/default.nix]: I did some digging, and it looks like `nuke-references` uses `$NIX_STORE`; I updated the PR to use that instead!

**Comment by rasendubi** [comment]
(triage) anything blocking this from being merged?

@dezgeg is your comment resolved?

**Comment by dotlambda** [comment]
conflicts with https://github.com/NixOS/nixpkgs/pull/51724

**Comment by FRidh** [comment]
Thanks. Pushed 2526f9f157465cfc66385ea3213ed833a3944c43 to staging.

**Pull request closed without merging**